src/modules/flt_scale.c

```diff
@@ -24,27 +24,29 @@ int init(int nextno, char* argstr) {
 		return 2;
 	}
 	free(argstr);
+
+	if (scale < 1) {
+		eprintf("flt_scale: Scale factor must be greater equal 1.\n");
+		return 1;
+	}
+
 	return 0;
 }
 
 int getx(void) {
-	assert(scale != 0);
 	return next->getx() / scale;
 }
 int gety(void) {
-	assert(scale != 0);
 	return next->gety() / scale;
 }
 
 int set(int x, int y, RGB *color) {
 	int px = 0;
 	int py = 0;
 	int ret = 0;
-	x = x * scale;
-	y = y * scale;
 	for (py = 0; py < scale; py++)
 		for (px = 0; px < scale; px++) {
-			ret = next->set(x + px, y + py, color);
+			ret = next->set(x * scale + px, y * scale + py, color);
```

Comment:
Ok this is a good point. But shouldn't the compiler be clever enough to see that it can optimize the multiplication by moving it outside the loop?
I just moved this inside, because it results in less code length.

Reply:
It'd probably be a good idea to keep the old version, just in case.